tests/Support/SupportCollectionTest.php

```diff
@@ -66,6 +68,90 @@ public function testFirstWithDefaultAndWithoutCallback($collection)
         $this->assertSame('default', $result);
     }
 
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testSoleReturnsFirstItemInCollectionIfOnlyOneExists($collection)
```

Comment:
Please also add tests in [SupportLazyCollectionIsLazyTest.php](https://github.com/laravel/framework/blob/8.x/tests/Support/SupportLazyCollectionIsLazyTest.php) ensuring this method is lazy.

Reply:
Yeah sure, I didn't realise those tests were there, so I'll add some to it